internal/kafka/internal/services/quota/ams_quota_service_test.go

```diff
@@ -1000,6 +1001,195 @@ func Test_AMSCheckQuota(t *testing.T) {
 	}
 }
 
+func Test_AMSReserveQuotaIfNotAlreadyReserved(t *testing.T) {
+	var amsDefaultKafkaConf = config.KafkaConfig{
+		Quota:                  config.NewKafkaQuotaConfig(),
+		SupportedInstanceTypes: test.NewAMSTestKafkaSupportedInstanceTypesConfig(),
+	}
+	type fields struct {
+		ocmClient   *ocm.ClientMock
+		kafkaConfig config.KafkaConfig
+	}
+	type args struct {
+		kafkaID                           string
+		kafkaInstanceType                 types.KafkaInstanceType
+		kafkaRequestMarketplace           string
+		kafkaRequestDesiredBillingModel   string
+		kafkaRequestBillingCloudAccountID string
+		kafkaRequestCloudProvider         string
+
+		wantsReserveCalled bool
+		wantsProductID     string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "Reserve quota for new cluster",
+			fields: fields{
+				kafkaConfig: amsDefaultKafkaConf,
+				ocmClient: &ocm.ClientMock{
+					FindSubscriptionsFunc: func(query string) ([]*v1.Subscription, error) {
+						return ams.FindSubscriptions(query), nil
```

Comment:
I understand the motivation of using the mock which is re-usability but it has the implication that it is effectively using some global state which means that changing the mock impacts all tests that use it. This makes in turn hard to perform changes when new needs arise for the mock, with effects of the change impacting several tests in the code

Reply:
I see your point. However, I put that mock in an internal package confined to the quota service test. The risk is limited enough to be acceptable.

Reply:
Even between two tests in a single file it is potentially fragile and I've seen that already happening in several places with package-global variables used between 1..N tests. I think we should aim to have per-test instances usually.
Let's keep this in mind when writing new tests where we need to introduce new elements